src/components/Nav/index.tsx

```diff
@@ -92,16 +94,37 @@ const Nav: FC<IProps> = ({ path }) => {
                   variant="ghost"
                   isSecondary
                   px={1.5}
+                  _hover={{
+                    transform: "rotate(30deg)", // Rotate the icon to 30 degrees on hover
+                    color: "primary.hover", // Change color to blue on hover
+                  }}
                   onClick={toggleColorMode}
                 ></IconButton>
+
                 <ButtonLink
+                  onMouseOver={() => setLanguagesHover(true)}
+                  onMouseOut={() => setLanguagesHover(false)}
                   to={`/languages/${fromPageParameter}`}
-                  leftIcon={<Icon as={MdLanguage} />}
                   variant="ghost"
                   isSecondary
                   px={1.5}
+                  _hover={{
+                    color: "primary.hover", // Change color on hover
```

Comment:
Unnecessary comment. The code is self-explanatory.
```suggestion
                    color: "primary.hover",
```